mino/minogrpc/controller/mod.go

```diff
@@ -87,7 +104,25 @@ func (m minimal) OnStart(ctx cli.Flags, inj node.Injector) error {
 
 	addr := minogrpc.ParseAddress("127.0.0.1", uint16(port))
 
-	o, err := minogrpc.NewMinogrpc(addr, rter)
+	var db kv.DB
+	err := inj.Resolve(&db)
+	if err != nil {
+		return xerrors.Errorf("injector: %v", err)
+	}
+
+	certs := certs.NewDiskStore(db, minogrpc.AddressFactory{})
+
+	key, err := m.getKey(ctx)
+	if err != nil {
+		return xerrors.Errorf("cert private key: %v", err)
+	}
+
+	opts := []minogrpc.Option{
+		minogrpc.WithStorage(certs),
+		minogrpc.WithCertificateKey(key, key.Public()),
+	}
```

Comment:
Just a design question: is this a common pattern in the code base ?

If not, what do you think about using a Fluent API? For configuration it's very natural. It might look like:

```
	opts := minogrpc.Options().
                   WithStorage(certs).
                   WithCertificateKey(key, key.Public())
```
and then you pass only one parameter, `opts`.

You could then call it with `opts.Apply(tmpl)` and all the details are hidden away from the library users. 

Then again, maybe this is actually just very idiomatic Go and I'm coming with baggage :-)

Reply:
I'm following this design pattern: https://www.sohamkamani.com/golang/options-pattern/

Two main reasons:
1. gRPC does too
2. I hate the formatting of chained functions in Go

Like always that's open to discussion but like you said it's a general pattern in the code base, so out of scope of this PR.

Reply:
Good enough for me :-) 

And yes, function chaining in Go looks horrible...